src/blueapi/service/authentication.py

```diff
@@ -54,14 +56,16 @@ def load_cache(self) -> Cache:
     def delete_cache(self) -> None:
         Path(self._file_path).unlink(missing_ok=True)
 
-    def _get_xdg_cache_dir(self) -> Path:
+    def _get_xdg_cache_path(self) -> Path:
         """
-        Return the XDG cache directory.
+        Return the XDG cache file path.
         """
-        cache_dir = os.environ.get("XDG_CACHE_HOME")
-        if not cache_dir:
-            cache_dir = os.path.expanduser(BLUEAPI_CACHE_LOCATION)
-        return Path(cache_dir)
+        cache_path = os.environ.get("XDG_CACHE_HOME")
+        if not cache_path:
+            cache_path = os.path.expanduser(BLUEAPI_CACHE_LOCATION)
+        elif os.path.isdir(cache_path):
+            cache_path = Path(cache_path) / "blueapi_cache"
+        return Path(cache_path)
```

Comment:
If `XDG_CACHE_HOME` is always a directory, would it be easier to make `BLUEAPI_CACHE_LOCATION` also a directory to keep the handling consistent? May be worth renaming to something like `DEFAULT_CACHE_DIR` as well. It's only used within blueapi so doesn't need the prefix and it makes it clearer that it might not be the path of the cache.

```suggestion
        cache_path = os.environ.get("XDG_CACHE_HOME", BLUEAPI_CACHE_LOCATION)
        return Path(cache_path).expanduser() / "blueapi_cache"
```

Reply:
It is a good assumption to make ...I made the changes ...
Thank you for the suggestion !